### PR TITLE
util/info.c: Improve error handling.

### PR DESCRIPTION
**Pull request by bturrubiates**
- Use strcasecmp so lowercase values will also be accepted.
- Add error handling to parsing functions. If a token is not recognized,
  then print a friendly error message and return -EINVAL. The program
  will exit with exit code EINVAL.

Signed-off-by: Ben Turrubiates \<bturrubi@cisco.com\>

### CODE REVIEW

File: util/info.c [hunk 1]
```diff
@@ -167,18 +177,28 @@ static uint32_t str2addr_format(char *inputstr)
...
 {
-	uint64_t flags = 0;
+	uint64_t value;
 	char *tok;
+	int ret;
 
-	for (tok = strtok(caps, "|"); tok != NULL; tok = strtok(NULL, "|"))
-		flags |= str2flag(tok);
+	for (tok = strtok(caps, "|"); tok != NULL; tok = strtok(NULL, "|")) {
+		ret = str2flag(tok, &value);
```
Comment (jsquyres) [inline util/info.c]: You're passing in a uint64t but not all those functions take that type.

Comment (jsquyres) [inline util/info.c]: Nope, I'm wrong. The ones that do not take uint 64 are not called by tokparse.

File: util/info.c [hunk 2]
```diff
@@ -133,32 +138,37 @@ static uint64_t str2cap(char *inputstr)
...
 	ORCASE(FI_DIRECTED_RECV);
 
-	return 0;
+	fprintf(stderr, "error: Unrecognized capability: %s\n", inputstr);
+
+	return -EINVAL;
 }
 
-static uint64_t str2mode(char *inputstr)
+static int str2mode(char *inputstr, uint64_t *value)
```
Comment (shefty) [inline util/info.c]: Mode is a 64-bit value - why change the function signature?

Comment (shefty) [inline util/info.c]: ah - never mind - got it

**Pull request merged**